src/com/gamecenter/TicTacToe.java

```diff
@@ -0,0 +1,91 @@
+package com.gamecenter;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class TicTacToe extends Board{
+    TicTacToeButtonListener tacListener = new TicTacToeButtonListener();
+
+    int alternate = 0;
+    int count = 0;
+
+    private TicTacToe(){
+        setTicTacToe_Listener();
+    }
+    public  class TicTacToeButtonListener implements ActionListener {
+        public void actionPerformed(ActionEvent e) {
+            JButton buttonClicked = (JButton) e.getSource();
+            if (alternate % 2 ==0){
+                buttonClicked.setText("X");
+            }else {
+                buttonClicked.setText("O");
+            }
+            if (checkWin()==true){
+                JOptionPane.showConfirmDialog(null,"Game-Over", "Gameover", JOptionPane.YES_NO_OPTION);
+                reSetButtons();
+            }
+            alternate++;
+            count++;
+        }
+    }
+    public void setTicTacToe_Listener(){
+        for (JButton button:buttons
+             ) {
+            button.addActionListener(tacListener);
+        }
+    }
+    public void reSetButtons(){
+        for (JButton button: buttons
+             ) {
```

Comment:
Look into instantiating a new game as opposed to clearing the board or clear the board and move counter